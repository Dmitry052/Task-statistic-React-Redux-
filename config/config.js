module.exports = {
  devlocal: {
    name: 'partner-cabinet',
    server: {
      log: {
        dir: './logs/'
      },
      port: 3000,
      api: {
        url: 'http://localhost:8082'
      }
    }
  },
  prod: {
    name: 'partner-cabinet',
    server: {
      log: {
        dir: './logs/'
      },
      port: 3000,
      api: {
        url: 'http://ico-api-app:80'
      }
    }
  }
};
