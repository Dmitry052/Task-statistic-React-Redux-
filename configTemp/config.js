module.exports = {
  development: {
    name: 'partner-cabinet',
    sentry: 'http://127.0.0.1',
    test: { url: 'http://localhost:3000' },
    server: {
      ssr: false,
      ssr_options: {
        payform_url: 'http://localhost:10101',
        localFormat: 'en-EN'
      },
      log: {
        dir: './logs/'
      },
      port: 3000,
      api: {
        url: 'http://192.168.250.254:6081'
      }
    }
  },
  stage: {
    name: 'partner-cabinet',
    redis: {
      host: 'redis',
      port: '6379'
    },
    sentry: 'http://127.0.0.1',
    server: {
      ssr: false,
      log: {
        dir: './logs/'
      },
      port: 3000,
      api: {
        url: 'http://stat-api:6081'
      }
    }
  },
  production: {
    name: 'partner-cabinet',
    sentry: 'http://127.0.0.1',
    test: { url: 'http://localhost:3000' },
    payform: {
      url: 'http://localhost:3000'
    },
    redis: {
      host: 'redis',
      port: '6379'
    },
    server: {
      ssr: false,
      ssr_options: {
        payform_url: 'http://localhost:10101',
        localFormat: 'en-EN'
      },
      log: {
        dir: './logs/'
      },
      port: 3000,
      api: {
        url: 'http://stat-api:6081'
      }
    }
  }
};
