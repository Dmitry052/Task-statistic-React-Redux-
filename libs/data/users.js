const users = require('./users.json');

const checkUser = {
  data: (login, pass) => {
    if (
      login !== undefined &&
      users[login] !== undefined &&
      pass === users[login].pass
    ) {
      return {
        auth: true,
        id: users[login].id,
        email: users[login].adviser_email,
        edit: users[login].edit,
        role: users[login].role
      };
    }
    return { status: false };
  }
};

module.exports = checkUser;
