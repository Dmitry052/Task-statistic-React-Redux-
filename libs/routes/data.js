const express = require('express');
const axios = require('axios');

const router = express.Router();

router.get('/daily', (req, res) => {
  const apiUrl = res.app.locals.api;

  axios
    .get(`${apiUrl}/land/stat/daily/`, {
      params: req.query
    })
    .then(response => {
      res.send({ status: 'OK', data: response.data });
    })
    .catch(err => {
      res.send(err);
    });
});

router.get('/campaign-stat', (req, res) => {
  const apiUrl = res.app.locals.api;

  axios
    .get(`${apiUrl}/land/stat/campaign-stat/`, {
      params: req.query
    })
    .then(response => {
      res.send({ status: 'OK', data: response.data });
    })
    .catch(err => {
      res.send(err);
    });
});

router.get('/campaignGeo', (req, res) => {
  const apiUrl = res.app.locals.api;

  axios
    .get(`${apiUrl}/land/stat/campaign-geo/`, {
      params: req.query
    })
    .then(response => {
      res.send({ status: 'OK', data: response.data });
    })
    .catch(err => {
      res.send(err);
    });
});

router.get('/adviser-list', (req, res) => {
  const apiUrl = res.app.locals.api;

  axios
    .get(`${apiUrl}/land/stat/adviser-list/`)
    .then(response => {
      res.send({ status: 'OK', data: response.data });
    })
    .catch(err => {
      res.send(err);
    });
});

router.get('/campaign-list', (req, res) => {
  const apiUrl = res.app.locals.api;

  axios
    .get(`${apiUrl}/land/stat/campaign-list`, {
      params: req.query
    })
    .then(response => {
      res.send({ status: 'OK', data: response.data });
    })
    .catch(err => {
      res.send(err);
    });
});

module.exports = router;
