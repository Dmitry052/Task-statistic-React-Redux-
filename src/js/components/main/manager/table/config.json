{
  "sourceData": {
    "url": "/manager/data/stat",
    "limit": 50,
    "offset": 0
  },
  "columns": [
    {
      "columnName": "Email",
      "field": "email",
      "type": "string"
    },
    {
      "columnName": "KYC create date",
      "field": "date_create",
      "type": "date"
    },
    {
      "columnName": "KYC expired date",
      "field": "date_confirm_expired",
      "type": "date"
    }
  ],
  "filters": [
    {
      "name": "to",
      "type": "date",
      "default": "-7",
      "displayname": "Create date"
    },
    {
      "name": "from",
      "type": "date",
      "default": "0",
      "displayname": "Expired date"
    },
    {
      "name": "email",
      "type": "text",
      "default": "",
      "displayname": "Email"
    },
    {
      "name": "status",
      "type": "select",
      "default": "NEW",
      "displayname": "Status",
      "value": ["NEW", "UPLOADED", "CONFIRMED", "CANCELED", "EXPIRED"]
    }
  ]
}
