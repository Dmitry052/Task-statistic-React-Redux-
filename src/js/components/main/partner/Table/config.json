{
  "sheets": [
    {
      "displayName": "Daily",
      "name": "daily",
      "sourceData": {
        "url": "/data/daily",
        "filters": [
          {
            "type": "date",
            "displayName": "From",
            "name": "from",
            "default": "-70"
          },
          {
            "type": "date",
            "displayName": "To",
            "name": "to",
            "default": "0"
          }
        ]
      },
      "fields": [
        {
          "field": "date",
          "displayName": "Date",
          "type": "string",
          "prefix": null,
          "postfix": null,
          "sortable": false,
          "total": { "text": "Total daily:" }
        },
        {
          "field": "hosts",
          "displayName": "Hosts",
          "type": "number",
          "prefix": null,
          "postfix": null,
          "prescision": 0,
          "sortable": true,
          "total": { "fn": "sum" }
        },
        {
          "field": "hits",
          "displayName": "Hits",
          "type": "number",
          "prefix": null,
          "postfix": null,
          "prescision": 0,
          "sortable": true,
          "total": { "fn": "sum" }
        }
      ]
    },
    {
      "displayName": "Campaign",
      "name": "campaign",
      "sourceData": {
        "url": "/data/campaign-stat",
        "filters": [
          {
            "type": "date",
            "displayName": "From",
            "name": "from",
            "default": "-20"
          },
          {
            "type": "date",
            "displayName": "To",
            "name": "to",
            "default": "0"
          },
          {
            "type": "select",
            "value": "campaign_id",
            "displayName": "campaign",
            "sourceData": {
              "url": "/data/campaign-list",
              "params": [
                {
                  "type": "global",
                  "name": "adviser_id"
                }
              ],
              "field": "campaignList"
            }
          }
        ]
      },
      "fields": [
        {
          "field": "campaign",
          "displayName": "Campaign",
          "type": "string",
          "prefix": null,
          "postfix": null,
          "sortable": false,
          "total": { "text": "Total campaign:" }
        },
        {
          "field": "hosts",
          "displayName": "Hosts",
          "type": "number",
          "prefix": null,
          "postfix": null,
          "prescision": 0,
          "sortable": true,
          "total": { "fn": "sum" }
        },
        {
          "field": "hits",
          "displayName": "Hits",
          "type": "number",
          "prefix": null,
          "postfix": null,
          "prescision": 0,
          "sortable": true,
          "total": { "fn": "sum" }
        }
      ]
    },
    {
      "displayName": "Campaign Geo",
      "name": "campaignGeo",
      "sourceData": {
        "url": "/data/campaignGeo",
        "filters": [
          {
            "type": "date",
            "displayName": "From",
            "name": "from",
            "default": "-30"
          },
          {
            "type": "date",
            "displayName": "To",
            "name": "to",
            "default": "0"
          }
        ]
      },
      "fields": [
        {
          "field": "country",
          "displayName": "Country",
          "type": "country",
          "prefix": null,
          "postfix": null,
          "sortable": false,
          "total": { "text": "Total country:" }
        },
        {
          "field": "hosts",
          "displayName": "Hosts",
          "type": "number",
          "prefix": null,
          "postfix": null,
          "prescision": 0,
          "sortable": true,
          "total": { "fn": "sum" }
        },
        {
          "field": "hits",
          "displayName": "Hits",
          "type": "number",
          "prefix": null,
          "postfix": null,
          "prescision": 0,
          "sortable": true,
          "total": { "fn": "sum" }
        }
      ]
    }
  ]
}
