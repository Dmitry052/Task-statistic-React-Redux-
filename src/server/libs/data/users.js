// @flow
const passwordHash = require('password-hash');
const users = require('./../../../../configTemp/users.js');

const checkUser = {
  data: (login: string, pass: string): Object => {
    if (
      login !== undefined &&
      users[login] !== undefined &&
      // pass === users[login].pass
      passwordHash.verify(pass, users[login].pass)
    ) {
      return {
        auth: true,
        id: users[login].id,
        email: users[login].adviser_email,
        edit: users[login].edit,
        user: login,
        role: users[login].role,
        route: users[login].route
      };
    }
    return { status: false };
  }
};

module.exports = checkUser;
