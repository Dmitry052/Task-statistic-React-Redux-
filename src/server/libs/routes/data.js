// @flow
import type { $Request, $Response, Router } from 'express';

const express = require('express');
const axios = require('axios');

const router: Router = express.Router();

router.get('/adviser-list', (req: $Request, res: $Response) => {
  const apiUrl = res.app.locals.api;

  axios
    .get(`${String(apiUrl)}/land/stat/adviser-list/`)
    .then(response => {
      res.send({ status: 'OK', data: response.data });
    })
    .catch(err => {
      res.send(err);
    });
});

router.get('/', (req: $Request, res: $Response) => {
  const apiUrl = res.app.locals.api;
  const { query }: { query: Object } = req;

  axios
    .get(`${String(apiUrl)}${String(query.url)}`, {
      params: JSON.parse(query.queryParams)
    })
    .then(response => {
      res.json({ status: 'OK', data: response.data });
    })
    .catch(err => {
      if (err.response) {
        res.json({ error: err.response.data });
      } else {
        res.json({ error: `Error connect ${String(query.url)} ` });
      }
    });
});

module.exports = router;
