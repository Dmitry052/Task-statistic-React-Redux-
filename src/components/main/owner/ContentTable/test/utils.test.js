import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import {
  formatDate,
  createHeader,
  createBodyTable,
  toTableFormat,
  totalData,
  formatTime,
  countryFormat
} from './../utils';
import data from './test.data.json';
import config from './test.config.json';

configure({ adapter: new Adapter() });

describe('Utils methods', () => {
  let contentTable;
  let bodyTable;
  let resTotalData;

  it('formatDate', () => {
    const time = '2018-04-16T11:07:27.000Z';
    expect(formatDate(time)).toBe('2018-04-16 14:07:27');
  });

  it('createHeader', () => {
    expect(createHeader(config.fields).length).toBe(3);
  });

  it('createBodyTable', () => {
    bodyTable = createBodyTable(config.fields);
    expect(bodyTable.length).toBe(10);
  });

  it('toTableFormat', () => {
    contentTable = toTableFormat([data]);
    expect(contentTable.length).toBe(6);
  });

  it('totalData', () => {
    resTotalData = totalData(contentTable, bodyTable);
    expect(Object.keys(resTotalData).length).toBe(10);
  });

  it('formatTime', () => {
    const now = formatDate(new Date()).split(' ')[0];
    expect(formatTime(0)).toBe(now);
  });

  it('countryFormat', () => {
    const tree = shallow(countryFormat('ru'));
    expect(toJson(tree)).toMatchSnapshot();
  });
});
