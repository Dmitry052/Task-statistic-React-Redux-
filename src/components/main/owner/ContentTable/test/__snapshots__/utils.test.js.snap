// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils methods countryFormat 1`] = `
<div>
  <ReactCountryFlag
    code="ru"
    svg={true}
  />
    
  ru
</div>
`;
